src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ReferencesPage/ImplicitUsingsValueProvider.cs

```diff
@@ -0,0 +1,200 @@
+                .Select(item =>
+                {
+                    string? isStaticMetadata = item.DirectMetadata.FirstOrDefault(metadata => metadata.Name.Equals("Static", StringComparison.Ordinal))?.EvaluatedValue;
+                    string? aliasMetadata = item.DirectMetadata.FirstOrDefault(metadata => metadata.Name.Equals("Alias", StringComparison.Ordinal))?.EvaluatedValue;
+
+                    return new ImplicitUsing(
+                        item.EvaluatedInclude,
+                        aliasMetadata,
+                        isStaticMetadata is not null && bool.TryParse(isStaticMetadata, out bool b) && b,
+                        item.IsImported);
+                }).ToList();
+        });
+
+        return usings;
+    }
+
+    private async Task<string> GetImplicitUsingsStringAsync()
+    {
+        List<ImplicitUsing> usings = await GetImplicitUsingsAsync();
+        return JsonConvert.SerializeObject(usings);
+    }
+    
+    public override Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
+    {
+        return GetImplicitUsingsStringAsync();
+    }
+
+    public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+    {
+        return GetImplicitUsingsStringAsync();
+    }
+
+    public override async Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+    {
+        await _threadingService.SwitchToUIThread();
+
+        if (JsonConvert.DeserializeObject(unevaluatedPropertyValue, typeof(List<ImplicitUsing>)) is not List<ImplicitUsing> usingsToSet)
+        {
+            return null;
+        }
+
+        usingsToSet.RemoveAll(usingToSet => usingToSet.IsReadOnly);
+
+        List<ImplicitUsing> existingUsings = await GetImplicitUsingsAsync();
+        
+        foreach (ImplicitUsing existingUsing in existingUsings)
+        {
+            if (!usingsToSet.Any(usingToSet => usingToSet.Include.Equals(existingUsing.Include)))
+            {
+                try
+                {
+                    await _projectAccessor.OpenProjectForWriteAsync(_configuredProject, project =>
+                    {
+                        project.RemoveItems(
+                            project.GetItems("Using").Where(i => string.Equals(existingUsing.Include, i.EvaluatedInclude, StringComparisons.ItemNames) && !i.IsImported)
+                        );
+                    });
+
+                }
+                catch (Exception ex)
+                {
+                    // if an import comes from a targets file, or else if there's a race condition we can't remove it. otherwise throw
+                    if (ex is not ArgumentException && ex is not InvalidOperationException)
```

Comment:
I'd prefer an explicit check that the item is not in the project file rather than depending on the lower layers of CPS throwing an exception here.